src/basics.md

```diff
@@ -502,6 +503,57 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-continuous-process-output]: #ex-continuous-process-output
+<a name="ex-continuous-process-output"></a>
+## Continuously process child process' outputs
+
+[![std-badge]][std] [![cat-os-badge]][cat-os]
+
+Contrary to [Run an external command and process
+stdout](#ex-parse-subprocess-output), while an external [`Command`] is
+running, process its standard output. A new pipe is created by
+[`Stdio::piped`] and it is read by a [`BufReader`]. Note, a buffered
+reader uses a buffer to reduce the number of I/O requests.
+
+The below recipe is equivalent to run the Unix shell command
+`journalctl | grep usb`.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+#
+use std::process::{Command, Stdio};
+use std::io::BufReader;
+use std::io::BufRead;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let stdout = Command::new("journalctl")
+        .stdout(Stdio::piped())
+        .spawn()?
+        .stdout
+        .ok_or_else(|| "Could not capture standard output.")?;
+
+    let reader = BufReader::new(stdout);
+
+    for line in reader.lines() {
+        let line = line?;
+        if line.find("usb").is_some() {
+            println!("{}", line);
+        }
+    }
+#
```

Comment:
the `Ok(())` might be a boilerplate but lets not hide it as it is necessary to compile the example when copy pasted